app/marketing/tasks.py

```diff
@@ -1,20 +1,99 @@
 from django.conf import settings
 
+import csv
+import os
 import marketing.stats as stats
 from app.services import RedisService
 from celery import app
 from celery.utils.log import get_task_logger
+from django.contrib.auth.models import User
+from django.utils import timezone
 from marketing.mails import new_bounty_daily as new_bounty_daily_email
-from marketing.mails import weekly_roundup as weekly_roundup_email
+from marketing.mails import weekly_roundup as weekly_roundup_email, send_mail
 from marketing.models import EmailSubscriber
 from marketing.utils import allowed_to_send_email
+from retail.emails import render_export_data_email, render_export_data_email_failed
 
 logger = get_task_logger(__name__)
 
 redis = RedisService().redis
 
 rate_limit = '300000/s' if settings.FLUSH_QUEUE or settings.MARKETING_FLUSH_QUEUE else settings.MARKETING_QUEUE_RATE_LIMIT
 
+def create_csv(export_type, profile, earnings):
+    path = f'app/assets/tmp/user-{export_type}/{profile}'
+    if not os.path.isdir(path):
+        os.makedirs(path)
+    name = f"{timezone.now().strftime('%Y_%m_%dT%H')}"
+    file_path = f'{path}/{name}.csv'
+
+    with open(file_path, 'w', encoding='utf-8') as earnings_csv:
+        writer = csv.writer(earnings_csv)
+        writer.writerow(['ID', 'Date', 'From', 'From Location', 'To', 'To Location', 'Type', 'Value In USD', 'TXID', 'Token Name', 'Token Value', 'URL'])
+        for earning in earnings:
+            writer.writerow([
+                earning.pk,
+                earning.created_on.strftime("%Y-%m-%dT%H:00:00"),
+                earning.from_profile.handle if earning.from_profile else '*',
+                earning.from_profile.data.get('location', 'Unknown') if earning.from_profile else 'Unknown',
+                earning.to_profile.handle,
+                earning.to_profile.data.get('location', 'Unknown'),
+                earning.source_type_human,
+                earning.value_usd,
+                earning.txid,
+                earning.token_name,
+                earning.token_value,
+                earning.url,
+            ])
+
+    return file_path
+    
+
+def send_csv(attachment, user_profile):
+    to_email = user_profile.user.email
+    from_email = settings.CONTACT_EMAIL
+    html, text, subject = render_export_data_email(user_profile=user_profile)
+    send_mail(
+        from_email,
+        to_email,
+        subject,
+        text,
+        html,
+        from_name=f"@{user_profile.handle}",
+        categories=['transactional'],
+        csv=attachment
+    )
+
+
+def send_download_failure_email(user_profile):
+    to_email = user_profile.user.email
+    from_email = settings.CONTACT_EMAIL
+    html, text, subject = render_export_data_email_failed(user_profile=user_profile)
+    send_mail(
+        from_email,
+        to_email,
+        subject,
+        text,
+        html,
+        from_name=f"@{user_profile.handle}",
+        categories=['transactional'],
+    )
```

Comment:
We normally only have celery decorated methods in a `tasks.py` file, could we move these over to `marketing/utils.py`?

Reply:
@gdixon I was working on this today, but moving those methods into `marketing/utils.py` caused circular dependencies. Specifically, `marketing.utils.py` is importing `marketing.mails.py` which imports `app.utils.py` which imports `marketing.utils.py`. I tried to resolve it but I ran out of time.